### PR TITLE
feat(sdk): Add ability to load secrets to volume in onprem

### PR DESCRIPTION
**Pull request by markwinter**
**Description of your changes:**
Currently `kfp.onprem.use_k8s_secret` only allows loading secrets to ENV variables.

This PR adds the ability to load them as a volume

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: sdk/python/kfp/onprem.py [hunk 1]
```diff
@@ -1,41 +1,61 @@
...
             train = train_op(...)
             train.apply(mount_pvc('claim-name', 'pipeline', '/mnt/pipeline'))
     """
+
     def _mount_pvc(task):
         from kubernetes import client as k8s_client
         # there can be other ops in a pipeline (e.g. ResourceOp, VolumeOp)
         # refer to #3906
-        if not hasattr(task, "add_volume") or not hasattr(task, "add_volume_mount"):
+        if not hasattr(task, "add_volume") or not hasattr(task,
```
Comment (NikeNano) [inline sdk/python/kfp/onprem.py:24]: nit: do this formatting here?

**Review by NikeNano — COMMENTED** [review]
Thanks for your contribution, there is something similar [here](https://github.com/kubeflow/pipelines/blob/6717434978f9c9d01d4c38914dc3e858a6a6c2fd/sdk/python/kfp/dsl/extensions/kubernetes.py#L18) as well in the dsl. Maybe we could collect this to one place? Might also be that this could be use full no matter onperm or not, what do  you think @markwinter?

**Comment by markwinter** [comment]
@NikeNano 
Thanks for reviewing.

That code looks very similar. It can probably be combined somehow. I noticed that kfp.azure is using the `use_secret` already. Will have a think about it

On another note, it looks like `kfp.dsl.extensions` is missing from the api docs https://kubeflow-pipelines.readthedocs.io/

**Comment by NikeNano** [comment]
> 
> On another note, it looks like `kfp.dsl.extensions` is missing from the api docs https://kubeflow-pipelines.readthedocs.io/

Good catch, we should fix that, I will take a look. 

>
>Will have a think about it

Awesome :)

File: sdk/python/kfp/onprem.py [hunk 2]
```diff
@@ -1,41 +1,60 @@
...
+            )
         )
+
     return _mount_pvc
 
 
-def use_k8s_secret(secret_name: str = 'k8s-secret', k8s_secret_key_to_env: Dict = {}):
+def use_k8s_secret(
+    secret_name: str = 'k8s-secret',
+    k8s_secret_key_to_env: Dict = {},
```
Comment (chensun) [inline sdk/python/kfp/onprem.py:46]: Understand this is from the existing code, but we should fix this: don't use mutable default arguments.
See: https://stackoverflow.com/questions/1132941/least-astonishment-and-the-mutable-default-argument

File: sdk/python/kfp/onprem.py [hunk 3]
```diff
@@ -1,41 +1,60 @@
...
 
-def mount_pvc(pvc_name='pipeline-claim', volume_name='pipeline', volume_mount_path='/mnt/pipeline'):
-    """Modifier function to apply to a Container Op to simplify volume, volume mount addition and
-    enable better reuse of volumes, volume claims across container ops.
+def mount_pvc(
+    pvc_name='pipeline-claim',
+    volume_name='pipeline',
+    volume_mount_path='/mnt/pipeline'
+):
+    """Modifier function to apply to a Container Op to simplify volume, volume
```
Comment (chensun) [inline sdk/python/kfp/onprem.py:9]: nit: Thanks for adding the docstring. Ideally, there should be a one-line short description, followed by a blank line, then the detailed description.
https://google.github.io/styleguide/pyguide.html#381-docstrings

**Comment by rimolive** [comment]
Closing this PR. No activity for more than a year.

/close

**Pull request closed without merging**